sor/src/test/java/com/bazaarvoice/emodb/sor/core/MinSplitSizeTest.java

```diff
@@ -61,4 +63,38 @@ public void testMinSplitAfterTimeout() {
         assertEquals(dataStore.getSplits("table", 10).size(), 20);
 
     }
+
+    @Test
+    public void testLocalResplitting() {
+        AstyanaxDataReaderDAO astyanaxDataReaderDAO = new AstyanaxDataReaderDAO(mock(PlacementCache.class), mock(ChangeEncoder.class), new MetricRegistry());
+
+        String[] minMaxResplits3Times = {
+                "0000000000000000000000000000000000000000",
+                "1fffffffffffffffffffffffffffffffffffffffe0",
+                "3fffffffffffffffffffffffffffffffffffffffc0",
+                "5fffffffffffffffffffffffffffffffffffffffa0",
+                "7fffffffffffffffffffffffffffffffffffffff80",
+                "9fffffffffffffffffffffffffffffffffffffff60",
+                "bfffffffffffffffffffffffffffffffffffffff40",
+                "dfffffffffffffffffffffffffffffffffffffff20",
+                "ffffffffffffffffffffffffffffffffffffffff"
+        };
+
+        List<Token> splits = astyanaxDataReaderDAO.resplitLocally(BOP20Partitioner.MINIMUM, BOP20Partitioner.MAXIMUM, 3);
+
+        assertEquals(splits.size(), minMaxResplits3Times.length);
+
+        for (int i =0; i < splits.size(); i++) {
+            assertEquals(splits.get(i).toString(), minMaxResplits3Times[i]);
+        }
+
+        // Number of tokens should be 2^10 + 1 = 1025, as 1025 consecutive tokens can form 1024 ranges.
+        assertEquals(astyanaxDataReaderDAO.resplitLocally(BOP20Partitioner.MINIMUM, BOP20Partitioner.MAXIMUM, 10).size(), 1025);
+
+        // Number of tokens should be 2^5 + 1 = 33, as 33 consecutive tokens can form 32 ranges.
+        assertEquals(astyanaxDataReaderDAO.resplitLocally(BOP20Partitioner.MINIMUM, BOP20Partitioner.MAXIMUM, 5).size(), 33);
+
+        // Number of tokens should be 2^0 + 1 = 2, as 2 consecutive tokens can form 1 range.
+        assertEquals(astyanaxDataReaderDAO.resplitLocally(BOP20Partitioner.MINIMUM, BOP20Partitioner.MAXIMUM, 0).size(), 2);
```

Comment:
Minor:  Since this is by far the most common use case it may be worth checking that the two tokens returned with `localResplits == 0` are just the original tokens, `BOP20Partitioner.MINIMUM` and `BOP20Partitioner.MAXIMUM`.

Reply:
👍